pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -308,6 +308,9 @@ def _training_step(
                 training_step_output = self.trainer.accelerator.training_step(step_kwargs)
                 self.trainer.accelerator.post_training_step()
 
+                # free memory
+                del step_kwargs
```

Comment:
@tchaton where does the batch get copied?

this here would only be necessary if we duplicate the batch, right?
If I look at the implementation of _build_kwargs, I can only see the reference being added to the dict.

Reply:
Yes, this doesn't have any impact. Can be removed !